### DISCUSSION TITLE
Object graph comparison - Is there a way to ensure that all properties on a subject are populated?

### DISCUSSION THREAD

**Discussion by dynamictulip** [fluentassertions/fluentassertions#2730] [Q&A]
I have a system to test which maps data from one type to another type, lets call them `Order` and `OrderDto`

In my scenario, `Order` has many more properties than `OrderDto` so I use `orderDto.Should().BeEquivalentTo(order, options => options.ExcludingMissingMembers());` to prevent an exception being thrown due to properties existing on `Order` which do not exist on `OrderDto`.

Some of the property names are slightly different between `Order` and `OrderDto` so I use `options.WithMapping<OrderDto>(e => e.Name, s => s.OrderName)` for each property that varies.

The problem is that I would like to ensure that every property that exists on `OrderDto` has been populated as it would be very easy to forget to add a `WithMapping` for a particular property, especially as things change in the future.

Is there a way to do this with object graph comparison like `options.IncludingMembers<OrderDto>()` or even an `orderDto.Should().NotHaveNullProperties()`?

**Comment by dennisdoomen** [comment]
Unfortunately no, although it shouldn't be too difficult to use a bit of Reflection to accomplish that. 

FWIW, I usually don't test the DTOs directly, but build HTTP API tests that send real (in-memory) HTTP requests and in which the assertion verifies the actual JSON that comes back. I see DTOs as implementation details.

**Reply by dynamictulip** [reply]
My real life scenario isn't DTOs (I pulled that example from the docs) but a legacy object which I have very little control over and just want a subset of the data from.

Not sure why I didn't think of reflection, thanks for the tip!
